util/src/main/java/io/kubernetes/client/Exec.java

```diff
@@ -170,20 +177,65 @@ public Process exec(
       throws ApiException, IOException {
     String path = makePath(namespace, name, command, container, stdin, tty);
 
-    WebSocketStreamHandler handler = new WebSocketStreamHandler();
-    ExecProcess exec = new ExecProcess(handler);
+    ExecProcess exec = new ExecProcess();
+    WebSocketStreamHandler handler = exec.getHandler();
     WebSockets.stream(path, "GET", apiClient, handler);
 
     return exec;
   }
 
+  static int parseExitCode(InputStream inputStream) {

```

Comment:
This returns zero on a number of edge cases which seems wrong.  I suggested some refactors...